### ISSUE TITLE
GPLv3 violation.

### ISSUE THREAD

**Issue by cathalferris** [GW8RDI/uSDXOpen#3]
This repository is in violation of the GPLv3 license.

Because you list the binaries as being released under GPLv3, you absolutely must have the source code available that creates the binaries that you make available - even if the code you used was itself open.

Not making the source code directly available is a GPLv3 violation. You do not get to enforce GPLv3 on others without fully respecting the license yourself.

To resolve this issue, you must put all of the source files up that were used to compile to the binaries that are listed here.

**Comment by GW8RDI** [comment]
Dear Mr Cathal Ferris, our lawyers have already tried to contact you and your employer regarding harassment.  They will continue to seek your address and take legal action against you.
Your claims are senseless, this GitHub page only contains source code.

**Comment by cathalferris** [comment]
:innocent:

Oh that's *so* cute. 
<pats-on-head>

--
Given your well-known recent mistakes with license interpretation, please make sure your "lawyers" :wink: have read and can understand [](https://www.gnu.org/licenses/gpl-faq.en.html) to make sure you don't "accidentally" fall foul when again not abiding by licenses.

**Comment by GW8RDI** [comment]
Cathal Ferris for a man of over 50 years of age, you are acting very immaturely.  I don't know what your problem is, but if you don't stop wasting our time you'll be blocked from this page.

**Comment by cathalferris** [comment]
It's good that you can admit you don't know what my problem is, because it's as simple as my having no problem at all. 
It is good that you can see that at least.

I'm not really sure why you would think I would have a problem, with you in particular anyway, as you're apparently not abusing licenses on my code at this point at least.

You do however appear to have a highly inappropriate level of antagonistic reactions to certain situations. Why *is* that? Conflict syndrome? High-conflict-personality? Both? 

*I* really am not wasting *your* time here.  
I'm not sure who you think the "our" in your statement refers to - I'm certainly not wasting my time at least.
If you're unwilling to lean from your mistakes, then you are probably wasting your time on more than this endeavour, but that's not on me.  
You don't have any contributors, so I couldn't possibly be wasting anybody else's time here.

As for me, I just like attempting to educate those that don't understand how licensing (especially the various flavours of Creative Commons and other Open Source licenses) works in the real world. It does result sometimes that those that abuse those licenses get overly-defensive through their misunderstandings of  things.

If you want to block me for having taken my valuable time to directly help you not make another public mistake regarding your use of licensed code, then that will have to be seen as abuse - by you. I've not given you a valid reason to block anyone after all.

Anyway, I hope you have a good licencing-compliant day, and I hope that it doesn't rain on you too much down there.
(here's a free Github hint - you can probably close this issue with a "wont-fix"...)

**Comment by chrisdebian** [comment]
Tidying up, can this be closed, now?

Chris
2E0FRU

**Comment by cathalferris** [comment]
No, I don't think this can be closed, as the situation I opened it for is not actually resolved.
The attempts of this fork to meet the license terms aren't actually meeting the license terms.

The readme of this project is also incredibly dishonest at the moment. Guido's text from the forked-from project does not link to the closed-group forums on Facebook that the originator of this fork rules with an iron fist. Guido's project links here: [uSDX Forum](https://groups.io/g/ucx). Maybe it would be correct to remove the edited text and instead have Guido's actual text instead of putting words in Guido's mouth..

This project still implies that this fork can be used on Manuel's project - but it is not licensed, and this is not addressed in this fork's documentation.

In short, this bug will be re-opened if closed, until the license terms are properly adhered to, and the documentation properly shows this in a clear and non-passive-aggressive manner..

(Also - it's not smart to threaten legal routes, especially if there's no follow-up. I was looking forwards to smacking any frivolous lawsuit back to where it belonged. The lack of a follow-up of any type does lend credence to my points. Such wonderful interactions with the community by the originator of this fork of Guido's code.)

**Comment by GW8RDI** [comment]
Cathal, one day I am going to meet you, and we can have a real discussion.
EU and US law is very clear in that owners of a product have the right to repair and maintain their own products exactly as they choose.  Legal cases where companies tried to force users to pay for their support were all thrown out.  Anyone interested can google "law right to repair".  You of course are not interested in people's rights.

**Comment by cathalferris** [comment]
Well isn’t that an interesting reply from the maintainer of this fork of Guido’s code?
Directly contravening the GitHub community guidelines for posts?

(Tl;dr: the bug was valid, and is still valid.)
--------------------------------

Let’s ignore the threats, the mistakes and the misunderstandings plainly seen in those responses for a little bit. I’ll bring the conversation back on-topic before I respond to that threat – I shall assume some measure of good faith for the moment even though it’s clearly not in play from the maintainer of this fork of Manuel’s and Guido’s good work with generating the μSDX from the SDX design of Hans Summer (G0UPL / AF7BF).

Threatening a bug reporter is not a good look, and generally will not be accepted. I will temporarily tolerate this poor behaviour in order to make sure that my points are available and understood.

So:
Let's see if this reply can put forward in a way that's clear and understandable. It would be good that offence is not misguidedly taken at something imagined or misunderstood or non-existent. It would be preferable to ensure that the tools and information are now available to the maintainer to be fully aware of and can therefore address, the points being made about the currently extant issues with this fork of Guido’s code. 

I am however aware that I can only do so much. No matter how clearly things can be laid out in front of a person, it is never possible to understand things *for* that person.

Let’s try anyway, and it’ll be seen if I am to be again wantonly misunderstood.

Guido's code is GPLv3. Good to see this is labelled as such in this fork, as it must be. This is what affords the maintainer of this fork of Guido’s code the ability to **_distribute_** any changes in this fork to his (and others’) code. Let’s understand there is an emphasis on "**_distribute_**" - and this will be returned to later on. It is suspected that there’s a core disconnect here between the maintainer of this fork’s understanding of licenses, and the terms of the code so licensed.

In the readme.md here, it’s very interesting to see that there’s no links to anything Manuel (DL2MAN) has done regarding his work on improving Hans’ hardware design, that is needed to execute this fork of Guido’s code. The fact that the work of others _is_ linked, does give the reader a certain impression about things..

A more relevant item to this bug is the interestingly insidious disingenuous distortion of Guido's message from the top of the readme for his project:
The maintainer appears to have tried to reproduce a portion of Guido’s readme.md on this fork's readme, possibly in an attempt to not breach the licensing terms:

“Reproduced” from the original project:- 
> [blocks of text]
> “Note: there is an active online forum discussing the uSDX controlled by DL2MAN who has closed his own USDX branch (tr)usdx. Other groups on FB include "USDX USDR Radios", "(tr)USDX Open".”
> 73, Guido [pe1nnz@amsat.org](mailto:pe1nnz@amsat.org)

The trouble is - this is not what Guido wrote.

This is what Guido _actually_ wrote:

> uSDX is a simple and experimental (Class-E driven) SSB and CW SDR transceiver.
> [blocks of text]
> Note: there is an active online forum discussing the uSDX here: [uSDX Forum](https://groups.io/g/ucx).
> 73, Guido [pe1nnz@amsat.org](mailto:pe1nnz@amsat.org)

Can a difference be seen? Can the import of this difference, and the import of its location within Guido’s quote be understood?

It would appear that the misquote of Guido’s readme has the (hopefully unintended) consequence of suggesting that Guido expressly supports this particular fork of his code – a  suggestion which appears to be inaccurate at best. There does not appear to be any independent direct expression of Guido’s explicit support for this fork’s edits of his work. The statements that others can work on his code - subject to the licensing on that code - is not to be taken as an explicit support of this fork, as those statements are nothing more than expression of the licensing terms.

Editing/changing someone else's work and not attributing the changes correctly; this may be an impression gathered by those paying attention to the “work” of the maintainer of this fork, however valid or not that impression may be, it is still an impression that can be garnered.

(side note: I always support the idea that it's perfectly fine to build upon the work of others – but it’s _only_ perfectly fine when one obeys the licensing terms of the code and licensing terms of the hardware that one is working on, and attribute correctly what was received when starting, when distributing either the results of the code edits or distributing the actual code edits. )

Reading this misquote of Guido in this fork’s readme - it may give a reader an impression that there's something underhand and possibly even slimy going on here; even if there may be no real basis to that impression. It's not hard to subvert that impression - just be truthful and correct in what is said, and comply fully with the licenses already agreed to abide by when obtaining the code. It's not hard to do it right, but it is hard to restore the trust lost when a person finds out about the implications of the misquote.

It would be suggested that future responses refrain from threats to the bugreporter or others, and that responses stay on topic and only address the concerns raised. It would not be to the benefit of this fork to have the maintainer give such a poor impression of their personal standards and possible lack of personal ethics.

--------------

Now, let’s address the previous threats and the unfortunate attempt to handwave away and distract others from the point I have raised regarding the standards in this fork. I’ll respond to the handwaving first before I respond to the threats.

It would be suggested that the maintainer actually read through and at least try to understand the terms and conditions that are imposed due to the GPLv3 license on this fork’s code. If the license were actually understood by the maintainer, it would be very unlikely that the response seen above would have been given. The content of the response does suggest that there is a significant cognitive dissonance between i) the license terms, and ii) what the maintainer thinks can be done with the code under this license. There are other licenses in play with the commits to this fork by the maintainer (still publicly available too) where code **_distributed_** by the maintainer contravened the license terms of use of hardware designed and powered by Manuel and Guido. 

Note well – the issue was the **_distribution_** of contravening code. Nobody cares about the perfectly valid and license-abiding coding to and executing such code on self-owned devices. The not-subtle distinction between i) generation and running of code intended for on your own device, and ii) generation of code to be allowable for license-abiding code **_distribution_**, appears to be one that the maintainer of this fork of Guido’s code is having significant difficulties with understanding.

The random handwave to “Right to Repair” is nothing more than an irrelevant distraction, as it has always been the case under GPLv3 that code can be changed and run on a self-owned device without any license repercussions. There’s never been any doubt about that, and there’s nothing in this bug that relates to anything relevant to running self-generated code on self-owned devices. 

The GPLv3 has relevance however when **_distributing_** code, such as by uploading to publicly accessible repositories such as this fork’s host.

Once that code is to be **_distributed,_** there are conditions that must be met, in order for that coder’s ability to be allowed to **_distribute_** that code. It’s all in black and white in the License, and should be easy enough to understand. There are plenty of helper guides on the Web to assist the understanding of those that have difficulty. One is only allowed to **_distribute_** the code that was originally obtained under the GPLv3, when meeting the terms of the GPLv3 when **_distributing_** code or compiled code onwards.

When feedback is provided to the maintainer of this site that their fork is in breach of a software license such as GPLv3 or CC BY-ND 4.0, it’s not smart to react with threats. Instead – be thankful that one has an opportunity to rectify the licensing mistake.

Some code that is in breach of a software license can still be seen by the public, can still be downloaded, and is still being made available (**_distributed_**) by the maintainer of this fork of Guido’s code.

To help the maintainer understand the predicament that they have caused themselves, let’s look at some commits that contain code that is expressly in violation of various licenses.

Here’s one commit (81a16e55150629ecbe212bc20f6252443bf427f9) with comments that are tragically incompetently written:

> “Herr Klärig (Ham callsign DL2MAN) has stated that his clients must not install other software on the radio that his chain of resellers offer on eBay, AliExpress and Amazon.”

It’s clear that the writer of this text has either not read, or misunderstands completely, the [Creative Commons Attribution-NoDerivatives 4.0 International License](http://creativecommons.org/licenses/by-nd/4.0/) that the hardware the code is written towards is available under. There’s not a lot of point in beating this particular dead horse, the real lawyers have spoken and the code is now harder to get – though it is publicly available still.

Then there’s the text that shows an inability to understand how a software open-source license works: 

> “The opinion of our lawyers is that Klärig is violating US, EU and UK consumer laws by restricting users rights to modify the product they purchased, the design of which is a clone of an entirely open source project”

There are at least three points on that particular statement:

- “entirely open source project” – the GPLv3 is not public-domain. There are restrictions in place that though people are allowed to freely obtain the code, freely change the code, freely run the code on their hardware, they do not have the rights to freely distribute the code unless they meet the GPLv3 terms. Maybe the writer of this text conflates “public-domain” with “open-source”? 
- It’s thoroughly disingenuous to state the following: “restricting users rights to modify the product they purchased”. This statement is patently untrue, and a clear misunderstanding of the reality of how open-source code licensing works. There’s absolutely no restriction on someone buying or building hardware, and running code they legitimately obtained and changed, on their own device. There are certainly restrictions on distribution of that modified code – as per the terms of how that code was obtained. Also, if a third party obtained the code, made changes, and then distributed that code to the hardware owner thus violating the license, then the recipient of that code has not legitimately obtained that code, and is not entitled to do anything with that code as the code is license-violating. This seems to be something that this fork’s maintainer has incredible difficulty with grasping.
- If these “laywers” are the same “lawyers” that have given other types of advice to this fork’s maintainer, it would be suggested that the maintainer try to talk to a real lawyer that didn’t find their license in a bag of Cornflakes or a two-Shilling Lucky Dip bag at the village fête. Hilariously poor advice given – at least if it was paid for there would be a chance of recompense. I’m still waiting for these “lawyers” to contact me, which is denying my ability to both counter-sue and confirm to some interested authorities the actual location to serve some “documentation”.


Speaking to the handwaving again: 

There's nothing in the license that restricts one from working on the code (if one has validly obtained the code) on your own devices. It has never in question that one can work on one’s own devices with code one has legitimately obtained and self-edited. Alluding to such matters when the bug report is about licence violations is utterly irrelevant to resolving this bug report, and should have been avoided in the responses. 



---

I have said before and I will continue to shine a light on those that take code from others, ignoring the license terms within, and DISTRIBUTING that code against the terms of the license.
I have a significant dislike of code-thieves, and especially code-thieves that try to sell the work of others as their own whether that's for cash or for kudos. I’m well aware of the fact that this dislike is not relevant to this particular bug, but it is important to me that people understand this, to see the position I come from. It’s not to benefit me personally, it’s to prevent people taking advantage of the good work of others.

I can also see that there could be a genuine mistake in what happened. How someone that violates a license publicly and getting called out on it, how they then handle the situation is the mark of their personal standards and ethics. The flip side of that is that a clear lack of personal ethics and a lack of (or wilful ignorance of) what a software or hardware license actually means will allow the community to have confirmation of that person’s clearly-questionable motives and methods.

Please do not conflate the concept of open-source licensed code with either copyright-free or public-domain code - as is continually suggested by some of the activities and responses seen in this fork.  
The optics of the previous responses here does make the maintainer of this fork of Guido’s code appear to be a little bit foolish, and far less trustworthy than would generally be desired.

The double-down on the mistake being promulgated by the maintainer of this fork of Guido's code, that there are "customers" of the (tru)SDX project, continues to bring amusement to the ham radio community.

The idiocy being expressed here does properly amuse me at times. The inability to use internet search engines, the inability to understand licenses, the inability to comprehend simple English at times - it would point to e.g. significant exposure to lead in the environment. Maybe the exhausts of the petrol-powered Army vehicles was leaking into the rear?

I see the edits to the "response". Question: Is the "we" in the phrase "as far as we know" the Royal "we", or the f20.9 "we", or even the f44.81 "we"? The answer determines the appropriate set of responses worth giving to that particular gem.

-------------
<offtopic>
As an aside - I would absolutely **_*love*_** to have you in front of me. Few things would give me as great a satisfaction.
If for no other reason to give myself a wonderful opportunity to help the unfortunate and to educate the uneducable. It’s always good to try to help others, even if those may not deserve that assistance and help.

It is always useful to note that I have absolutely no fear from blowhards, pompous buffoons, shysters, and small-time conmen. Thus, I also have absolutely no fear due to you or anything you think you can do. What's the phrase that comes to mind anytime I see empty threats such as hinted above? Ah yes: "All mouth and no trousers".

It’s not a good look to publicly threaten (even if emptily) legal responses (or any other type of threat implicit or otherwise), when receiving perfectly valid feedback on code and how it’s being distributed. It’s an even worse look to try to double-down on the threats. Doing that kind of thing makes it easy for the real lawyers to win big against the attempted bully. 

If I did deign to meet with you,  I suspect the gap between the _"discussion"_ you thought you'd have and the reality of meeting me and both seeing and understanding that it is far from a keyboard warrior that I am, and that I would say the same things to someone's face that I would say online. I fear such a meeting though would unfortunately be to your detriment and to the amusement of myself and others.
</offtopic>

-----------------------

Back on-topic; the bug was valid, and is still valid.
-----------------------

Also – readers should always be very careful with reading tone from text. The reader takes the tone they expect from text, not that which was intended.. 
With this response I intend the tone to be accurate and neutral. Any other tone taken is at the risk of that reader.
